airflow/api_connexion/endpoints/dag_run_endpoint.py

```diff
@@ -122,6 +122,7 @@ def _fetch_dag_runs(
     limit,
     offset,
 ):
+    total_entries = query.count()
```

Comment:
This would filter before any date limits are applied, and thus would get an incorrect count.

Reply:
Good count. I think the original behavior included a bug. @mik-laj can you confirm that the count should be taken after filtering by date?

Reply:
`total_entries` should return the number of elements that are available on all the page of the current request.

Reply:
Actually, I think we should maintain the existing behavior here and do the count prior to the date filtering. Just to keep scope limited. If we decide that the count should take place after the date filtering, let's make that behavioral change in a separate PR, just to reduce scope for this one somewhat.

Reply:
Can you create a issues about it?

Reply:
I've created an issue to change how the total_entries are counted. #11074